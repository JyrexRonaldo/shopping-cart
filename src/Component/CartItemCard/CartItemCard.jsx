import styles from "./CartItemCard.module.css"

function CartItemCard({ imgURL, title, price, quantity }) {
  return (
    <div className={styles.cartItemCard}>
      <div><img src={imgURL} alt="" /></div>
      <div>
        <p>{title}</p>
        <p>Price: {`$${price}`}</p>
        <p>Quantity: {quantity}</p>
        <p>Total: {`$${price * quantity}`}</p>
      </div>
      <div><button type="button">Remove</button></div>
    </div>
  );
}

export default CartItemCard