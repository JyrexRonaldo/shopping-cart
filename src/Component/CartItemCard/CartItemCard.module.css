.cartItemCard {
    width: 600px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    gap: 20px;

}

.cartItemCard img {
    width: 80px;
    height: auto;
    object-fit: contain;
  }

  .cartItemCard > div:first-of-type {
    width: 300px;
    height: auto;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .cartItemCard > div:nth-of-type(2) {
    width: 300px;
    
  }