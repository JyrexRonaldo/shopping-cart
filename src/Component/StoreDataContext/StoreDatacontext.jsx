import { createContext } from "react";

const StoreDataContext = createContext(null);

export default StoreDataContext