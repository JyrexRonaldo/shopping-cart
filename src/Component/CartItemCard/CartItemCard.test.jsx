import { render } from "@testing-library/react";
import CartItemCard from "./CartItemCard";
import { describe, it, expect } from "vitest";
import StoreDataContext from "../StoreDataContext/StoreDataContext";


describe("CartItemCard component", () => {
  it("it displays elements with props supplied", () => {
    const mockItemData = {
      imgURl: "image of item",
      title: "cool shirt",
      price: "429",
      quantity: 2,
      itemId: 1,
    };

    const contextData = {};

    const { container } = render(
      <StoreDataContext.Provider value={contextData}>
        <CartItemCard
          imgURL={mockItemData.imgURl}
          title={mockItemData.title}
          price={mockItemData.price}
          quantity={2}
          itemId={1}
        />
      </StoreDataContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
