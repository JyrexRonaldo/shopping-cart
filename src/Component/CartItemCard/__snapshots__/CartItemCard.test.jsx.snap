// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CartItemCard component > it displays elements with props supplied 1`] = `
<div>
  <div
    class="_cartItemCard_3a5faf"
  >
    <div>
      <img
        alt=""
        src="image of item"
      />
    </div>
    <div>
      <p>
        cool shirt
      </p>
      <p>
        Price: 
        $429
      </p>
      <p>
        Quantity: 
        2
      </p>
      <p>
        Total: 
        $858
      </p>
    </div>
    <div>
      <button
        type="button"
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;
